### DISCUSSION TITLE
C++ constexpr named const

### DISCUSSION THREAD

**Discussion by EchoPouet** [carbon-language/carbon-lang#1887] [Language design]
There is `var` and `let` but will there also be `const` which is the equivalent of `constexpr` in C++ ?

If so, will it be able to hold the contents of a file that will be read at compile time?

**Comment by geoffromer** [comment]
> There is `var` and `let` but will there also be `const` which is the equivalent of `constexpr` in C++ ?

We'll certainly need to support the use cases covered by `constexpr` in C++, but we haven't yet decided exactly what that will look like. `let` already covers some of that space, as do `:!` and `template :!` bindings.

> 
> If so, will it be able to hold the contents of a file that will be read at compile time?

I expect we'll eventually support reading files at compile time, but we don't have a design for that yet.
